src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3037,8 +3037,29 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             if (ASRUtils::expr_value(arg) != nullptr) {
                 char *c = ASR::down_cast<ASR::StringConstant_t>(
                                     ASRUtils::expr_value(arg))->m_s;
+                int ival = 0;
+                try {
+                    char *ch = c;
+                    if ((ch)[0] == '-') {
+                        ch++;
+                    }
+                    while (*ch) {
+                        char chr = *ch;
+                        switch (chr) {
+                            case '.' :
+                                throw std::string(""); break;
+                            default :
+                                if (chr < '0' || chr > '9')
+                                    throw std::string(""); break;
+                        }
+                        ch++;
+                    }
+                    ival = std::stoi(c);
+                } catch(...) {
+                    throw SemanticError("invalid literal for int() with base 10: '"+ std::string(c) + "'", arg->base.loc);
+                }
```

Comment:
Overall this PR looks good, tests are good, it fixes a real issue. My last comment is to the code here --- we should not use `catch(...)` similar as in Python you should not use a catch all `except`. We should not use exceptions at all in a code like this. Rather let's rewrite this. You can use `break` to exit a while loop.

Would you please mind improving the code above? 

After that I think we can merge this.

Reply:
You can use `if` condition to raise a `SemanticError`.

Reply:
I have used `if` statements now and it passes all tests locally ,kindly check it out .